drivers/vtencent/util.go

```diff
@@ -0,0 +1,303 @@
+	"github.com/alist-org/alist/v3/pkg/utils"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/go-resty/resty/v2"
+)
+
+func (d *Vtencent) request(url, method string, callback base.ReqCallback, resp interface{}) ([]byte, error) {
+	req := base.RestyClient.R()
+	req.SetHeaders(map[string]string{
+		"cookie":       d.Cookie,
+		"content-type": "application/json",
+		"origin":       d.conf.origin,
+		"referer":      d.conf.referer,
+	})
+	if callback != nil {
+		callback(req)
+	} else {
+		req.SetBody("{}")
+	}
+	if resp != nil {
+		req.SetResult(resp)
+	}
+	res, err := req.Execute(method, url)
+	if err != nil {
+		return nil, err
+	}
+	code := utils.Json.Get(res.Body(), "Code").ToString()
+	if code != "Success" {
+		switch code {
+		case "AuthFailure.SessionInvalid":
+			if err != nil {
+				return nil, errors.New(code)
+			}
+		default:
+			return nil, errors.New(code)
+		}
+		return d.request(url, method, callback, resp)
+	}
+	return res.Body(), nil
+}
+
+func (d *Vtencent) ugcRequest(url, method string, callback base.ReqCallback, resp interface{}) ([]byte, error) {
+	req := base.RestyClient.R()
+	req.SetHeaders(map[string]string{
+		"cookie":       d.Cookie,
+		"content-type": "application/json",
+		"origin":       d.conf.origin,
+		"referer":      d.conf.referer,
+	})
+	if callback != nil {
+		callback(req)
+	} else {
+		req.SetBody("{}")
+	}
+	if resp != nil {
+		req.SetResult(resp)
+	}
+	res, err := req.Execute(method, url)
+	if err != nil {
+		return nil, err
+	}
+	code := utils.Json.Get(res.Body(), "Code").ToInt()
+	if code != 0 {
+		message := utils.Json.Get(res.Body(), "message").ToString()
+		if len(message) == 0 {
+			message = utils.Json.Get(res.Body(), "msg").ToString()
+		}
+		return nil, errors.New(message)
+	}
+	return res.Body(), nil
+}
+
+func (d *Vtencent) LoadUser() (string, error) {
+	api := "https://api.vs.tencent.com/SaaS/Account/DescribeAccount"
+	res, err := d.request(api, http.MethodPost, func(req *resty.Request) {}, nil)
+	if err != nil {
+		return "", err
+	}
+	return utils.Json.Get(res, "Data", "TfUid").ToString(), nil
+}
+
+func (d *Vtencent) GetFiles(dirId string) ([]File, error) {
+	api := "https://api.vs.tencent.com/PaaS/Material/SearchResource"
+	form := fmt.Sprintf(`{
+		"Text":"",
+		"Text":"",
+		"Offset":0,
+		"Limit":20000,
+		"Sort":{"Field":"%s","Order":"%s"},
+		"CreateTimeRanges":[],
+		"MaterialTypes":[],
+		"ReviewStatuses":[],
+		"Tags":[],
+		"SearchScopes":[{"Owner":{"Type":"PERSON","Id":"%s"},"ClassId":%s,"SearchOneDepth":true}]
+	}`, d.Addition.OrderBy, d.Addition.OrderDirection, d.TfUid, dirId)
+	var resps RspFiles
+	rsp, err := d.request(api, http.MethodPost, func(req *resty.Request) {
+		req.SetBody(form).ForceContentType("application/json")
+	}, &resps)
+	if err != nil {
+	}
+	if err := json.Unmarshal(rsp, &resps); err != nil {
+		return []File{}, err
+	}
```

Comment:
[> 响应头ContentType有问题，所以后面json不解析。可以加上这个解决
](https://github.com/alist-org/alist/pull/5526/files#r1398076408)

这里你理解错了,我的意识是可用把代码改成这样,不需要额外json.Unmarshal. 只需要传入request就行

```suggestion
	var resps RspFiles
	rsp, err := d.request(api, http.MethodPost, func(req *resty.Request) {
		req.SetBody(form).ForceContentType("application/json")
	}, &resps)
	if err != nil {
	    return []File{}, err 
	}
```